### PR TITLE
Determine CONDA_ENV_SYSROOT from environment

### PR DESCRIPTION
**Pull request by pearu**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

See discussions in:
https://github.com/conda-forge/staged-recipes/pull/9225/files#r330386001

### CODE REVIEW

**Comment by pearu** [comment]
@conda-forge-admin, please rerender

File: recipe/install_nvcc.sh [hunk 1]
```diff
@@ -64,11 +64,20 @@ export CFLAGS="\${CFLAGS} -I\${CUDA_HOME}/include"
 export CPPFLAGS="\${CPPFLAGS} -I\${CUDA_HOME}/include"
 export CXXFLAGS="\${CXXFLAGS} -I\${CUDA_HOME}/include"
 
+echo "ACTIVATE NVCC"
+env
+
+if [[ "\${CONDA_BUILD:-0}" = "1" ]]
+then
+  CONDA_ENV_SYSROOT=\${PREFIX}/\${HOST}/sysroot
```
Comment (isuruf) [inline recipe/install_nvcc.sh]: You could make it `CONDA_ENV_SYSROOT=${PREFIX}/${HOST}/sysroot` (without the escapes) and conda will fix the paths when installing this package.

Comment (pearu) [reply recipe/install_nvcc.sh]: I tried that but after installing nvcc_linux-64 locally, the nvcc activate script contains:
```
CONDA_ENV_SYSROOT=/opt/anaconda1anaconda2anaconda3/x86_64-conda_cos6-linux-gnu/sysroot
```
which seems wrong.

Comment (pearu) [reply recipe/install_nvcc.sh]: This is in `/home/pearu/mconda3/pkgs/nvcc_linux-64-10.1-hf484d3e_7/etc/conda/activate.d/nvcc_linux-64_activate.sh`.
But
```
$ ls -ls /home/pearu/mconda3/envs/test-nvcc2/x86_64-conda_cos6-linux-gnu/sysroot/lib/libcuda.so 
0 lrwxrwxrwx 1 pearu pearu 44 nov    3 19:33 /home/pearu/mconda3/envs/test-nvcc2/x86_64-conda_cos6-linux-gnu/sysroot/lib/libcuda.so -> /home/pearu/cuda/10.1/lib64/stubs/libcuda.so
```
which seems correct.

Comment (pearu) [reply recipe/install_nvcc.sh]: I trust that the `PREFIX` part is fixed but will also the `HOST` part be fixed?

Comment (isuruf) [reply recipe/install_nvcc.sh]: conda will replace that prefix with the correct one when that package is installed into a conda environment

Comment (isuruf) [reply recipe/install_nvcc.sh]: `HOST` part will not be fixed, but I think that's fine for now.

File: recipe/install_nvcc.sh [hunk 2]
```diff
@@ -64,11 +64,14 @@ export CFLAGS="\${CFLAGS} -I\${CUDA_HOME}/include"
 export CPPFLAGS="\${CPPFLAGS} -I\${CUDA_HOME}/include"
 export CXXFLAGS="\${CXXFLAGS} -I\${CUDA_HOME}/include"
 
+CONDA_ENV_SYSROOT=${PREFIX}/${HOST}/sysroot
```
Comment (isuruf) [inline recipe/install_nvcc.sh]: Can you create this directory if it doesn't exist?

Comment (pearu) [reply recipe/install_nvcc.sh]: The subsequent `mkdir -p "\${CONDA_ENV_SYSROOT}/lib"` handles this.

File: recipe/install_nvcc.sh (hunk 1 above)
Comment (pearu) [reply recipe/install_nvcc.sh]: Resolved (modulo assuming that `HOST` part is correct).

**Comment by kkraus14** [comment]
@jakirkham I think you should review this as I know you had opinions regarding how we detect the sysroot.

**Comment by jakirkham** [comment]
Thanks for the work here. Sorry for the delay. Am a bit slammed ATM. So likely won't have time to look until some time midweek next week. Sorry again for the delay.

**Comment by pearu** [comment]
The recent `gcc_linux-64` provides `CONDA_BUILD_SYSROOT` (PR https://github.com/conda-forge/ctng-compiler-activation-feedstock/pull/9), so we don't need to determine `CONDA_ENV_SYSROOT` in the nvcc installation script anymore.

@isuruf @jakirkham , please review.

**Comment by jakirkham** [comment]
Thanks @pearu! 😄

**Pull request merged**